scripts/maint/rename_c_identifier.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python3
+
+"""
+Helpful script to replace one or more C identifiers, and optionally
+generate a commit message explaining what happened.
+"""
+
+import argparse
+import fileinput
+import os
+import re
+import shlex
+import subprocess
+import sys
+import tempfile
+
+TOPDIR="src"
+
+def is_c_file(fn):
+    """
+       Return true iff fn is the name of a C file.
+
+       >>> is_c_file("a/b/module.c")
+       True
+       >>> is_c_file("a/b/module.h")
+       True
+       >>> is_c_file("a/b/module.c~")
+       False
+       >>> is_c_file("a/b/.module.c")
+       False
+       >>> is_c_file("a/b/module.cpp")
+       False
+    """
+    fn = os.path.split(fn)[1]
+    if fn.startswith("."):
+        return False
+    ext = os.path.splitext(fn)[1]
+    return ext in {".c",".h",".i","inc"}
+
+def list_c_files(topdir=TOPDIR):
+    """
+       Use git to list all the C files under version control.
+
+       >>> lst = list(list_c_files())
+       >>> "src/core/mainloop/mainloop.c" in lst
+       True
+       >>> "src/core/mainloop/twiddledeedoo.c" in lst
+       False
+    """
+    proc = subprocess.Popen(
+        ["git", "ls-tree", "--name-only", "-r", "HEAD", topdir],
+        stdout=subprocess.PIPE,
+        encoding="utf-8")
+    for line in proc.stdout.readlines():
+        line = line.strip()
+        if is_c_file(line):
+            yield line
+
+class Rewriter:
+    """
+       A rewriter applies a series of word-by-word replacements, in
+       sequence.  Replacements only happen at word boundaries.
```

Comment:
What is a word?
Can you be clearer about which symbols and numbers are parts of words?
Can you show examples?

Reply:
"A word is defined as a sequence of alphanumeric or underscore characters"
"the precise set of characters deemed to be alphanumeric depends on the values of the UNICODE and LOCALE flags"
https://docs.python.org/2/library/re.html#regular-expression-syntax

Reply:
Strictly, this doesn't match C identifier syntax, which is `[a-zA-Z_][a-zA-Z_0-9]*`. But that's probably ok.

Reply:
c9a484361d2b452098cc8bdbfa71fbce22b9ce90

Reply:
21c9c85da45fa2b65567e39cd8dc586a1643f93a